File: python/剑指offer/面试题58 - I. 翻转单词顺序.py
class Solution:
    def reverseWords(self, s: str) -> str:
        alist = [i for i in s.strip().split(" ") if i != ""]
        return " ".join(alist[::-1])

s = Solution()
test = "a good   example"
print(s.reverseWords(test))

File: go/剑指offer/面试题58 - I. 翻转单词顺序.go
package main

import "strings"

func reverseWords(s string) string {
	s = strings.TrimSpace(s)
	arr := strings.Split(s , " ")
	var stack []string
	for i:= len(arr)-1;i>=0;i-- {
		if arr[i] != ""{
			stack = append(stack, arr[i])
		}
	}
	return strings.Join(stack," ")
}
